src/main/java/backend/resource/TurboLabel.java

```diff
@@ -211,4 +229,136 @@ public int hashCode() {
+                .map(TurboLabel::getActualName)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of label names that matches a query
+     * Uses partial matching and matching is case-insensitive
+     * @param repoLabels
+     * @param nameQuery
+     * @return
+     */
+    public static List<String> filterByPartialName(List<String> repoLabels, String nameQuery) {
+        return repoLabels
+                .stream()
+                .filter(name -> Utility.containsIgnoreCase(
+                    new TurboLabel("", name).getSimpleName(), nameQuery))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of label names that belongs to a group and matches a query
+     * Uses partial matching and matching is case-insensitive
+     * @param repoLabels
+     * @param groupQuery
+     * @return
+     */
+    public static List<String> filterByPartialGroupName(List<String> repoLabels, String groupQuery) {
+        if (groupQuery.isEmpty()) return repoLabels;
+
+        return repoLabels
+                .stream()
+                .filter(name -> {
+                    TurboLabel newLabel = new TurboLabel("", name);
+                    if (newLabel.isInGroup()) {
+                        return Utility.containsIgnoreCase(newLabel.getGroupName(), groupQuery);
+                    }
+                    return false;
+                })
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the first label that matches the keyword
+     * i.e. the label's group contains keyword's group and label's name contains keyword's name
+     * @param repoLabels
+     * @param keyword
+     * @return
+     */
+    public static String getMatchedLabelName(List<String> repoLabels, String keyword) {
+        TurboLabel query = new TurboLabel("", keyword);
+
+        List<String> newMatchedLabels = new ArrayList<>();
+        newMatchedLabels.addAll(repoLabels);
+        newMatchedLabels = filterByPartialName(newMatchedLabels, query.getSimpleName());
+        newMatchedLabels = filterByPartialGroupName(newMatchedLabels, query.getGroupName());
+        return newMatchedLabels.get(0);
+    }
+
+    /**
+     * Returns true if there is exactly 1 matching label for keyword
+     *
+     * A label is matching if:
+     * the label's group contains keyword's group and label's name contains keyword's name
+     * @param repoLabels
+     * @param keyword
+     * @return
+     */
+    public static boolean hasExactlyOneMatchedLabel(List<String> repoLabels, String keyword) {
+        TurboLabel query = new TurboLabel("", keyword);
+
+        List<String> newMatchedLabels = new ArrayList<>();
+        newMatchedLabels.addAll(repoLabels);
+        newMatchedLabels = filterByPartialName(newMatchedLabels, query.getSimpleName());
+        newMatchedLabels = filterByPartialGroupName(newMatchedLabels, query.getGroupName());
+        return newMatchedLabels.size() == 1;
```

Comment:
As mentioned earlier today, implementation of this method can be reduced to 
`getMatchedLabelName(...).size() == 1` if the `getMatchedLabelName` is changed to return all matched labels.
Even if you don't do that, do extract the duplicated code between the two methods.